cg/apps/mip/base.py

```diff
@@ -38,51 +39,49 @@ class MipStartError(Exception):
 class MipAPI:
     """ Group MIP specific functionality """
 
-    def __init__(self, script, pipeline, logger=logging.getLogger(__name__)):
+    def __init__(self, script, pipeline, conda_env, logger=logging.getLogger(__name__)):
         """Initialize MIP command line interface."""
         self.script = script
         self.pipeline = pipeline
+        self.conda_env = conda_env
         self.logger = logger
 
     def run(self, config, case, **kwargs):
-        """Execute the pipeline."""
+        """Execute the workflow"""
         command = self.build_command(config, case, **kwargs)
-        self.logger.debug(" ".join(command))
+        self.logger.debug(SPACE.join(command))
         process = self.execute(command)
-        process.wait()
         success = 0
         if process.returncode != success:
             raise MipStartError("error running analysis, check the output")
         return process
 
     def build_command(self, config, case, **kwargs):
-        """Builds the command to execute MIP."""
+        """Builds the command to execute MIP"""
         command = [
+            f"bash -c 'source activate {self.conda_env};",
             self.script,
             self.pipeline,
             case,
             CLI_OPTIONS["config"]["option"],
             config,
         ]
         for key, value in kwargs.items():
-            # enable passing in flags as "False" - shouldn't add command
             if value:
+                # Cg to mip options mapping
                 command.append(CLI_OPTIONS[key]["option"])
-                if value is True:
-                    command.append(CLI_OPTIONS[key].get("default", "1"))
-                else:
+                # append value for non-flags
+                if value is not True:
                     command.append(value)
+        command.append(SINGLE_QUOTE)
         return command
 
     @classmethod
-    def execute(cls, command):
-        """Start a new MIP run."""
-
+    def execute(cls, command: str):
```

Comment:
This class should use the [`Process`](https://github.com/Clinical-Genomics/cg/blob/master/cg/utils/commands.py) interface that is used in CG when communicating via shell

Reply:
Done